starboard/shared/starboard/player/filter/audio_discard_duration_tracker.cc

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/shared/starboard/player/filter/audio_discard_duration_tracker.h"
+
+#include <algorithm>
+
+#include "starboard/common/log.h"
+#include "starboard/shared/starboard/media/media_util.h"
+
+namespace starboard {
+namespace shared {
+namespace starboard {
+namespace player {
+namespace filter {
+using shared::starboard::media::AudioSampleInfo;
+
+void AudioDiscardDurationTracker::CacheDiscardDuration(
+    const scoped_refptr<InputBuffer>& input_buffer,
```

Comment:
Just to double check, if there is an audio input buffer with absolute time range 24.95 seconds to 25.05 seconds.  Now we are seeking to 25.00 seconds, which may form a partial audio input buffer with 50 ms discarded from the front.  In such case, is the timestamp of the input buffer passed in already adjusted to 25.00 seconds, or still 24.95 seconds?